src/lib.rs

```diff
@@ -69,11 +78,34 @@ impl Plugin for XpbdPlugin {
             .register_type::<InvInertia>()
             .register_type::<LocalCom>();
 
-        // Add stages
-        app.add_stage_before(
-            CoreStage::Update,
-            FixedUpdateStage,
-            SystemStage::parallel().with_run_criteria(run_criteria),
+        let mut xpbd_schedule = Schedule::default();
+
+        xpbd_schedule.configure_sets(
+            (
+                PhysicsSet::Prepare,
+                PhysicsSet::BroadPhase,
+                PhysicsSet::Substeps,
+            )
+                .chain(),
+        );
+
+        app.add_schedule(XpbdSchedule, xpbd_schedule);
+
+        app.add_schedule(XpbdSubstepSchedule, Schedule::default());
+
+        // Add system set for running physics schedule
+        app.configure_set(
+            FixedUpdateSet
+                .before(CoreSet::Update)
+                .in_base_set(CoreSet::PreUpdate),
+        );
+        app.add_system(run_physics_schedule.in_set(FixedUpdateSet));
+
+        // move to plugin?
+        app.add_system(
+            run_substep_schedule
+                .in_set(PhysicsSet::Substeps)
+                .in_schedule(XpbdSchedule),
         );
```

Comment:
Do you mean that we would have a plugin like `SubstepPlugin` that runs and initializes the `XpbdSubstepSchedule`? We could even move the `NumSubsteps` resource into the same file to group substepping code a bit more.

Would the substepping schedule's plugins (currently just `IntegratorPlugin` and `SolverPlugin`) be initialized in the substepping plugin as well, or should they stay in the `XpbdPlugin`?

Reply:
Yeah, that was the question. I started doing that, then I felt like it's kind of odd to split the overall xpbd algorithm across multiple files, and decided to keep it in `lib.rs`. I feel like when it's just adding two plugins, it's not really worth the extra indirection (adds unnecessary complexity imo).

Removing the todo now, let me know if I should do the refactor now, or we could perhaps do it in a follow-up pr.

Reply:
Yeah I think we can keep it in `lib.rs` for now to avoid splitting the overall structure. It can always be changed later if there's a good reason.